src/Common/HostResolvePool.cpp

```diff
@@ -237,10 +254,22 @@ void HostResolver::updateImpl(Poco::Timestamp now, std::vector<Poco::Net::IPAddr
     }
 
     for (auto & rec : merged)
-        if (rec.failed && rec.fail_time < last_effective_resolve)
-            rec.failed = false;
+    {
```

Comment:
Here I adjust new counter `banned_count`. class `Rec` is unaware about metrics, as a result that code does not belong Rec's method.